base/condition.jl

```diff
@@ -90,7 +90,8 @@ function _wait2(c::GenericCondition, waiter::Task)
         # set it to be sticky.
         # XXX: Ideally we would be able to unset this
         ct.sticky = true
-        tid = Threads.threadid()
+        tid = Threads.rawthreadid()
+        @assert Threads.threadpoolid() == Threads.threadpoolid(waiter)
```

Comment:
Does this hold always given that the public APIs are used correctly?

Reply:
Yeah, this seems potentially wrong. First, the branch should be checking `rawthreadid(waiter) == 0`. Then, the assertion (if desired) should be moved after the ccall (and only if it succeeds), to ensure that we placed the task into the right threadpool.